PolicyDecisionPoint/src/main/java/it/cnr/iit/ucs/pdp/PolicyDecisionPoint.java

```diff
@@ -268,31 +238,21 @@ private AbstractResult evaluateContext( EvaluationCtx context, PolicyFinder poli
         PolicyFinderResult finderResult = policyFinder.findPolicy( context );
 
         if( finderResult.notApplicable() ) {
-            return ResultFactory.getFactory()
-                .getResult( AbstractResult.DECISION_NOT_APPLICABLE, context );
-        }
-        if( finderResult.indeterminate() ) {
-            return ResultFactory.getFactory().getResult(
-                AbstractResult.DECISION_INDETERMINATE, finderResult.getStatus(),
-                context );
-        }
-        // we found a valid policy, list all found policies if XACML 3.0
-        if( context instanceof XACML3EvaluationCtx
-                && ( (RequestCtx) context.getRequestCtx() ).isReturnPolicyIdList() ) {
+            return ResultFactory.getFactory().getResult( AbstractResult.DECISION_NOT_APPLICABLE, context );
+        } else if( finderResult.indeterminate() ) {
+            return ResultFactory.getFactory().getResult( AbstractResult.DECISION_INDETERMINATE, finderResult.getStatus(), context );
+        } else if( context instanceof XACML3EvaluationCtx && ( (RequestCtx) context.getRequestCtx() ).isReturnPolicyIdList() ) {
+            // we found a valid policy, list all found policies if XACML 3.0
             Set<PolicyReference> references = new HashSet<>();
             processPolicyReferences( finderResult.getPolicy(), references );
             ( (XACML3EvaluationCtx) context ).setPolicyReferences( references );
         }
-
-        // we can perform the evaluation
         return finderResult.getPolicy().evaluate( context );
     }
 
-    private void processPolicyReferences( AbstractPolicy policy,
-            Set<PolicyReference> references ) {
+    private void processPolicyReferences( AbstractPolicy policy, Set<PolicyReference> references ) {
         if( policy instanceof Policy ) {
```

Comment:
:-) how original.  Reading it is funny "if policy is policy" !!
How about renaming it to abstractPolicy

Reply:
ahah yeah ok not my code :P

Reply:
done